metric_learn/_util.py

```diff
@@ -678,17 +678,20 @@ def _initialize_metric_mahalanobis(input, init='identity', random_state=None,
 
   random_state = check_random_state(random_state)
   M = init
-  if isinstance(init, np.ndarray):
-    s, u = scipy.linalg.eigh(init)
-    init_is_definite = _check_sdp_from_eigen(s)
+  if isinstance(M, np.ndarray):
+    w, V = eigh(M, check_finite=False)

```

Comment:
why did you turn off `check_finite`? I think it is better to keep it

Reply:
`check_array() ` does it previously, so it shouldn't be necessary to do it again.

Reply:
Indeed!